tests/lib/smf/src/test_lib_flat_smf.c

```diff
@@ -204,7 +214,7 @@ static void state_d_entry(void *obj)
 
 static void state_d_run(void *obj)
 {
-	/* Do nothing */
+	smf_set_terminate(obj, 1);
```

Comment:
I'd prefer a macro for the terminate value here - `STATE_D_TERMINATE_VALUE`.
It would also be an improvement to change the other termination calls to use unique terminate values.